starter_code/src/movies.js

```diff
@@ -1,15 +1,2122 @@
+    rate: 8
+  },
+  {
+    title: "The Best Years of Our Lives",
+    year: 1946,
+    director: "William Wyler",
+    duration: "2h 50min",
+    genre: ["Drama", "Romance", "War"],
+    rate: 8.1
+  },
+  {
+    title: "Mou gaan dou",
+    year: 2002,
+    director: "Wai-Keung Lau",
+    duration: "1h 41min",
+    genre: ["Crime", "Drama", "Mystery", "Thriller"],
+    rate: 8.1
+  },
+  {
+    title: "Paris, Texas",
+    year: 1984,
+    director: "Wim Wenders",
+    duration: "2h 25min",
+    genre: ["Drama"],
+    rate: 8.1
+  },
+  {
+    title: "The Help",
+    year: 2011,
+    director: "Tate Taylor",
+    duration: "2h 26min",
+    genre: ["Drama"],
+    rate: 8.1
+  },
+  {
+    title: "Faa yeung nin wa",
+    year: 2000,
+    director: "Kar-Wai Wong",
+    duration: "1h 38min",
+    genre: ["Drama", "Romance"],
+    rate: 8.1
+  },
+  {
+    title: "Sholay",
+    year: 1975,
+    director: "Ramesh Sippy",
+    duration: "3h 18min",
+    genre: ["Action", "Adventure", "Comedy", "Drama", "Musical", "Thriller"],
+    rate: 8.2
+  },
+  {
+    title: "Beauty and the Beast",
+    year: 1991,
+    director: "Gary Trousdale",
+    duration: "1h 24min",
+    genre: ["Animation", "Family", "Fantasy", "Musical", "Romance"],
+    rate: 8
+  },
+  {
+    title: "La battaglia di Algeri",
+    year: 1966,
+    director: "Gillo Pontecorvo",
+    duration: "2h 1min",
+    genre: ["Drama", "War"],
+    rate: 8.1
+  },
+  {
+    title: "Ah-ga-ssi",
+    year: 2016,
+    director: "Chan-wook Park",
+    duration: "2h 24min",
+    genre: ["Crime", "Drama", "Mystery", "Romance", "Thriller"],
+    rate: 8.1
+  },
+  {
+    title: "Pirates of the Caribbean: The Curse of the Black Pearl",
+    year: 2003,
+    director: "Gore Verbinski",
+    duration: "2h 23min",
+    genre: ["Action", "Adventure", "Fantasy"],
+    rate: 8
+  },
+  {
+    title: "PK",
+    year: 2014,
+    director: "Rajkumar Hirani",
+    duration: "2h 33min",
+    genre: ["Comedy", "Drama", "Fantasy", "Sci-Fi"],
+    rate: 8.2
+  },
+  {
+    title: "Dog Day Afternoon",
+    year: 1975,
+    director: "Sidney Lumet",
+    duration: "2h 5min",
+    genre: ["Biography", "Crime", "Drama", "Thriller"],
+    rate: 8
+  },
+  {
+    title: "Dead Poets Society",
+    year: 1989,
+    director: "Peter Weir",
+    duration: "2h 8min",
+    genre: ["Comedy", "Drama"],
+    rate: 8
+  }
+];
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+// function orderByYear(array) {
+//     const newArray = [...array];
+//     let sort = 0;
+//     return newArray.sort((a, b) => {
+//         if (a.year === b.year) {
+//             return a.title < b.title ? -1 : 1;
+//         } else {
+//             return a.year - b.year;
+//         }
+//     });
+// }
+
+function orderByYear(array) {
+  const newArray = [...array];
+  return newArray.sort((a, b) =>
+    a.year === b.year ? (a.title < b.title ? -1 : 1) : a.year - b.year
+  );
+}
+
```

Comment:
If you want to make it short you could do this: 
```
function orderByYear(array) {
  return [...array].sort((a, b) => (a.year === b.year ? a.title.localeCompare(b.title) : a.year - b.year));
}
```
localeCompare is a function on the string prototype that returns either 1 , - 1 or 0, which is what the sort function is expecting :D
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare